python/tk_desktop/extensions/__init__.py

```diff
@@ -10,9 +10,58 @@
 
 import sys
 
-if sys.platform == "darwin":
-    from .darwin import *
-elif sys.platform == "win32":
-    from .win32 import *
-elif sys.platform.startswith("linux"):
-    from .linux import *
+import sgtk
+
+logger = sgtk.platform.get_logger(__name__)
+
+use_mocked_osutils = True
+if sys.platform.startswith("linux"):
+    try:
+        import PySide  # noqa
+    except Exception:
+        # PySide 2 build of desktop does not require help
+        # making the application go in the foreground/background.
+        pass
+    else:
+        # We're in a PySide 1 version of Shotgun Desktop, so try
+        # to import the old osutils
+        try:
+            from .linux_python2_qt4 import osutils
+
+            use_mocked_osutils = False
+        except Exception as e:
+            logger.warning("Could not load osutils: %s", e, exc_info=True)
+elif sys.platform == "darwin":
+    # On macOS, the osutils are required to make the app move to the foreground
+    # or background in certain cases.
+    try:
+        if sys.version_info[0] == 2:
```

Comment:
Why on Mac is the version dependent on python version, but on Linux its on PySide version?

Reply:
See comments above (which I've just added)